File: ROOT/WEB-INF/application/app/models/GenerationSession.java
package models;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Set;
import java.util.TreeSet;

import play.data.validation.Min;
import play.data.validation.Max;
import play.data.validation.Required;

public class GenerationSession implements Serializable  {

	public String rows;

	public String columns;

	public Set<String> cellValues = new TreeSet<String>();

	public String[][] matrix;

	public String delimiter = ";";

	public String fileName = "file.csv";

	public GenerationSession() {
		super();
	}

	public Step1Params getStep1Params() {
		Step1Params params = new Step1Params();
		params.columns = columns;
		params.rows = rows;
		return params;
	}

	public Step2Params getStep2Params() {
		Step2Params params = new Step2Params();
		return params;
	}

	public Step4Params getStep4Params() {
		Step4Params params = new Step4Params();
		params.delimiter = delimiter;
		params.fileName = fileName;
		return params;
	}

	public void reallocateMatrix() {

		String[][] newMatrix = new String[Integer.parseInt(rows)][Integer.parseInt(columns)];

		if (matrix != null) {
			int rowMin = matrix.length < newMatrix.length ? matrix.length : newMatrix.length;
			int colMin = matrix[0].length < newMatrix[0].length ? matrix[0].length : newMatrix[0].length;
			for (int i = 0; i < rowMin; i++) {
				for (int j = 0; j < colMin; j++) {
					newMatrix[i][j] = matrix[i][j];
				}
			}
		}

		matrix = newMatrix;

	}

	@Override
	public String toString() {
		return "GenerationSession [rows=" + rows + ", columns=" + columns
				+ ", cellValues=" + cellValues + "]";
	}






}

File: csv_generator2/app/models/GenerationSession.scala
package models

import scala.collection.{mutable, SortedSet}

case class GenerationSession(rowsNoI: Int, columnsNoI: Int, cellValuesI: SortedSet[String],matrixI: Array[Array[String]], delimiterI: String, filenameI: String) {

  def this() = this(0, 0, mutable.TreeSet[String]()(Ordering[String]), Array.ofDim[String](0, 0), ";", "file.csv" )

  var rowsNo: Int = rowsNoI
  var columnsNo: Int = columnsNoI
  var cellValues: SortedSet[String] = cellValuesI
  var matrix: Array[Array[String]] = matrixI
  var delimiter: String = delimiterI
  var filename: String = filenameI

/*  def apply() = {
    this.rowsNo = 0
    this.columnsNo = 0
    this.cellValues = mutable.TreeSet[String]()(Ordering[String])
    this.matrix =  Array.ofDim[String](rowsNo, columnsNo)
    this.delimiter = ";"
    this.filename = "file.csv"
  }*/

/*  def GenerationSession(rowsNo: Int, columnsNo: Int, cellValues: SortedSet[String],matrix: Array[Array[String]], delimiter: String, filename: String) = {
    this.rowsNo = rowsNo
    this.columnsNo = columnsNo
    this.cellValues = mutable.TreeSet[String]()(Ordering[String])
    this.matrix =  Array.ofDim[String](rowsNo, columnsNo)
    this.delimiter = delimiter
    this.filename = filename
  }*/



  def reallocateMatrix = {
    val newMatrix = Array.ofDim[String](rowsNo, columnsNo)

    if (matrix.length > 0 && matrix(0).length > 0) {

      var rowMin = newMatrix.length
      if (matrix.length < newMatrix.length) {
        rowMin = matrix.length
      }
      var colMin = newMatrix(0).length
      if (matrix(0).length < newMatrix(0).length) {
        colMin = matrix(0).length
      }

      for (i <- 0.to(rowMin - 1)) {
        for (j <- 0.to(colMin - 1)) {
          newMatrix(i)(j) = matrix(i)(j)
        }
      }

    }

    matrix = newMatrix
  }

  def getRows: List[Row] = {
    var rows = List[Row]()
    for (i <- 0.to(rowsNo - 1)) {
      rows = rows :+ Row(matrix(i).toList)
    }
    rows
  }

  def setRows(rows: List[Row]) = {
    for (i <- 0.to(rows.length - 1)) {
      matrix(i) = rows(i).cells.toArray
    }
  }


  /*
  	public void reallocateMatrix() {

		String[][] newMatrix = new String[Integer.parseInt(rows)][Integer.parseInt(columns)];

		if (matrix != null) {
			int rowMin = matrix.length < newMatrix.length ? matrix.length : newMatrix.length;
			int colMin = matrix[0].length < newMatrix[0].length ? matrix[0].length : newMatrix[0].length;
			for (int i = 0; i < rowMin; i++) {
				for (int j = 0; j < colMin; j++) {
					newMatrix[i][j] = matrix[i][j];
				}
			}
		}

		matrix = newMatrix;

	}

*/


}

